pkg/kinds/catalog.go

```diff
@@ -0,0 +1,107 @@
+package kinds
+
+import (
+	"context"
+
+	"github.com/grafana/kindsys"
+)
+
+type Catalog interface {
+	CoreKinds(ctx context.Context) []kindsys.Core
+	CustomKinds(ctx context.Context) []kindsys.Custom
+	ComposableKinds(ctx context.Context) []kindsys.Composable
+	AllKinds(ctx context.Context) []kindsys.Kind
+	AllProviders(ctx context.Context) []kindsys.Provider
+	Register(ctx context.Context, p *kindsys.Provider)
+	Unregister(ctx context.Context, p *kindsys.Provider)
+}
```

Comment:
What should the relationship be between this and what we have in `pkg/registry/corekind`?